app/models/cart.rb

```diff
@@ -0,0 +1,29 @@
+class Cart
+  attr_reader :contents
+
+  def initialize(session_cart)
+    @contents = session_cart || {}
+  end
+
+  def add_item(item_id, quantity = 1)
+    contents[item_id.to_s] ||= 0
+    contents[item_id.to_s]  += (1 * quantity)
+  end
+
+  def total_count
+    contents.values.sum
+  end
+
+  def count_of(item_id)
+    contents[item_id.to_s]
+  end
+
+  def items
+    item_ids = contents.keys
+    Item.where(id: item_ids)
+  end
+
+  def total_price
```

Comment:
@tmikeschu 

When calculating total_price should we also be using the quantity in the contents hash? I may be misunderstanding this but it looks like items is only counting unique items. If I'm not off my rocker would something like this work?

`contents.keys.reduce(0) {|sum, id| sum += Item.where(id).total_price * contents[id] }`

or

`items.reduce(0) {|sum, item| sum += item.total_price * contents[item.id] }`